scripts/iscsictl.py

```diff
@@ -0,0 +1,310 @@
+
+        # Default configuration uses incoming autentication
+        self.ssh.echo('IncomingUser user passwd', '>> /etc/ietd.conf')
+        self.ssh.echo('Target iqn.2015-01.qa.cloud.suse.de:id01',
+                      '>> /etc/ietd.conf')
+        self.ssh.echo('-e', '"\tLun 0 Path=%s"' % self.loop,
+                      '>> /etc/ietd.conf')
+
+        # Persist and start the service
+        self.service('iscsitarget', True)
+        self.ssh.rciscsitarget('restart')
+
+        # Check if the device is exported
+        result = str(self.ssh.cat('/proc/net/iet/volume'))
+        if 'iqn.2015-01.qa.cloud.suse.de:id01' not in result:
+            raise Exception('Unable to deploy the iSCSI target')
+
+
+class Initiator(ISCSI):
+    """Define and manage an iSCSI initiator node."""
+
+    # For now, we are going to use the discovery option for iSCSI to
+    # populate the database.  This simplify the deployment of a basic
+    # iSCSI scenario, with a single target point and multiple
+    # initiators.
+
+    def __init__(self, ssh, target_ip, mount):
+        """Initialize the Initiator instance with an ip and a mount point."""
+        super(Initiator, self).__init__(ssh)
+        self.target_ip = target_ip
+        self.mount = mount
+
+    def deploy(self, mkfs=False):
+        """Deploy, configure and persist an iSCSI initiator."""
+        self.zypper('open-iscsi')
+
+        # Default configuration only takes care of autentication
+        lines = (
+            'node.session.auth.authmethod = CHAP',
+            'node.session.auth.username = user',
+            'node.session.auth.password = passwd',
+            'discovery.sendtargets.auth.authmethod = CHAP',
+            'discovery.sendtargets.auth.username = user',
+            'discovery.sendtargets.auth.password = passwd',
+        )
+        for line in lines:
+            self.ssh.echo(line, '>> /etc/iscsid.conf')
+
+        # Persist and start the service
+        self.service('open-iscsi', True)
+        self.ssh.sh('rcopen-iscsi', 'restart')
+
+        # Discovery and login
+        result = self.ssh.iscsiadm('-m', 'discovery', '--type=st',
+                                   '--portal=%s' % self.target_ip)
+        name = 'iqn.2015-01.qa.cloud.suse.de:id01'
+        if name not in result:
+            raise Exception('Target %s not found' % name)
+        self.ssh.iscsiadm('-m', 'node', '-n', name, '--login')
+
+        if mkfs:
+            self.ssh.mkfs('/dev/sda1')
+
+        self.ssh.mkdir('-p', self.mount)
+        self.ssh.mount('/dev/sda1', self.mount)
+
+    def umount(self):
+        """Umount the shared device."""
+        self.ssh.umount(self.mount)
+
+        name = 'iqn.2015-01.qa.cloud.suse.de:id01'
+        self.ssh.iscsiadm('-m', 'node', '-n', name, '--logout')
+
+
+def test():
+    """Testing against mkcloud."""
+    # Adming node is going to be the target
+    admin = SSH('root', '192.168.124.10', 'linux')
```

Comment:
OK, I just thought it might be generalized for broader usage (this also applies for my other comment)